preprocessing/reformat_data.py

```diff
@@ -0,0 +1,61 @@
+import pandas as pd
+import os
+
+
+def snap_reviews(rating: int):
+    """
+    Function to snap a rating (1-5) to either negative, neutral or positive.
+    :param rating: star rating
+    :return: sentiment as a string
+    """
+    if rating < 3:
+        return "negative"
+    elif rating > 3:
+        return "positive"
+    return "neutral"
+
+
+def format_data(path: str):
+    """
+    Format data from csv file and return dataframe.
+    :param path: path to review csv file
+    :return: formatted data as dataframe
+    """
+    df = pd.read_csv(filepath_or_buffer=path, names=["star_rating", "headline", "body"])
+
+    # From stars to sentiment
+    df['star_rating'] = df['star_rating'].apply(snap_reviews)
+
+    # Discard neutral reviews
+    df = df[df['star_rating'] != "neutral"]
+
+    # Change header
+    df.set_axis(['sentiment', 'headline', 'body'], axis='columns')
+    return df
+
+
+def main():
+    # parent directory
+    parent = os.path.join(os.getcwd(), os.pardir)
+    parent = os.path.abspath(parent)
+
+    # Define paths of test and training
+    test_path = os.path.join(parent, "dataset", "test.csv")
```

Comment:
This gives me a `FileNotFoundError` on my Mac, as in VS Code `os.cwd()` returns the root folder for the project workspace, not the folder in which the module itself (in this case `reformat_data.py`) is located. See the traceback below:

```[Errno 2] No such file or directory: '/Users/Will/Documents/ECE 539 Final Project/dataset/test.csv'
  File "/Users/Will/Documents/ECE 539 Final Project/ECE539-Final-Project/preprocessing/reformat_data.py", line 24, in format_data
    df = pd.read_csv(filepath_or_buffer=path, names=["star_rating", "headline", "body"])
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Will/Documents/ECE 539 Final Project/ECE539-Final-Project/preprocessing/reformat_data.py", line 47, in main
    formatted_test = format_data(test_path)
                     ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Will/Documents/ECE 539 Final Project/ECE539-Final-Project/preprocessing/reformat_data.py", line 61, in <module>
    main()
FileNotFoundError: [Errno 2] No such file or directory: '/Users/Will/Documents/ECE 539 Final Project/dataset/test.csv'```

Reply:
I think we might need a more robust solution to directory management, sadly I don't know what that would be though.

Reply:
Tried replacing it with os.path.curdir, if that doesn't work on your mac, we'll just use pathlib instead

Reply:
I think this is an environment thing too, as I am getting the same error using VS Code on my Windows PC. I think the issue is that Code uses the root project directory as its working directory, not the parent folder of the file being executed; I was able to get the code running by replacing line 42 of `reformat_data.py` with the following:
```parent = os.path.curdir```

I think the only way to get consistent behavior across execution environments is to call main() from a script which is in the root project directory (either a separate script or `reformat_data.py` itself if we move it to the top level). We might have a decent number of files kicking around in our root directory if we put everything there, but I think that with proper moduling (combining your and Leah's preprocessing steps into one file, for example) we can mitigate the issue.